types/promise-retry/index.d.ts

```diff
@@ -0,0 +1,25 @@
+// Type definitions for promise-retry 1.1
+// Project: https://github.com/IndigoUnited/node-promise-retry
+// Definitions by: Jamie Birch <https://github.com/shirakaba>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { WrapOptions } from "retry";
+/**
+ * A function that is retryable, by having implicitly-bound params for both an error handler and an attempt number.
+ *
+ * @param retry The retry callback upon any rejection. Essentially throws the error on in the form of a { retried: err }
+ * wrapper, and tags it with a 'code' field of value "EPROMISERETRY" so that it is recognised as needing retrying. Call
+ * this from the catch() block when you want to retry a rejected attempt.
+ * @param attempt The number of the attempt.
+ * @returns A Promise for anything (eg. a HTTP response).
+ */
+type RetryableFn = ((retry: (error: any) => never, attempt: number) => Promise<any>);
+/**
+ * Wrap all functions of the object with retry. The params can be entered in either order, just like in the original library.
```

Comment:
If these can be entered in either order, I assume that they're exclusive (i.e. you can't pass in a function and a function, or options and another  options bag), and that in the case of options going first, the function is *required*. Should these be overloads instead?

```ts
declare function promiseRetry(retryableFn: RetryableFn, options?: WrapOptions): Promise<any>;
declare function promiseRetry(options: WrapOptions, retryableFn: RetryableFn): Promise<any>;
```

Reply:
Ah yes, you're quite right. I've pushed this change now.